### PR TITLE
[Default Configuration Part 5]: Move default configuration related classes to aws-core

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
- Move default configuration related classes from `sdk-core` to `aws-core` 
- Remove `defaultsMode` from `ClientOverrideConfiguration`.
- Add `defaultsMode` to AwsClientBuilder
- Minor refactoring on the configuration resolution in `AwsDefaultClientBuilder`

### CODE REVIEW

File: core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsClientBuilder.java [hunk 1]
```diff
@@ -64,4 +65,24 @@
...
+     *     <li>Check the "defaults_mode" profile file property.</li>
+     *     <li>Check "aws.defaultsMode" system property.</li>
+     *     <li>Check the "AWS_DEFAULTS_MODE" environment variable.</li>
+     * </ol>
+     *
+     * @param defaultsMode the defaultsMode to use
+     * @return This object for method chaining.
+     * @see DefaultsMode
+     */
+    default BuilderT  defaultsMode(DefaultsMode defaultsMode) {
```
Comment (Bennett-Lynch) [inline core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsClientBuilder.java:85]: Hyper-nit: 1 extra space

File: core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java [hunk 2]
```diff
@@ -333,4 +342,14 @@ public final void setCredentialsProvider(AwsCredentialsProvider credentialsProvi
         return Arrays.asList(new HelpfulUnknownHostExceptionInterceptor(),
                              new EventStreamInitialRequestInterceptor());
     }
+
+    @Override
+    public final BuilderT  defaultsMode(DefaultsMode defaultsMode) {
```
Comment (Bennett-Lynch) [inline core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java:347]: 1 extra space

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkSystemSetting.java [hunk 3]
```diff
@@ -170,8 +169,7 @@
     AWS_MAX_ATTEMPTS("aws.maxAttempts", null),
 
     /**
-     * Which {@link DefaultsMode} to use, case insensitive
-     * @see DefaultsMode
+     * Which {@code DefaultsMode} to use, case insensitive
      */
     AWS_DEFAULTS_MODE("aws.defaultsMode", null),
```
Comment (Bennett-Lynch) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkSystemSetting.java:174]: Does this still belong in `sdk-core`?

**Review by Bennett-Lynch — APPROVED** [review]

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkSystemSetting.java (hunk 3 above)
Comment (zoewangg) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkSystemSetting.java:174]: We don't actually have an `AwsSystemSetting`. AWS related system properties such as `AWS_ACCESS_KEY_ID` are in this class already, so I kept it here to be consistent.

**Review by Bennett-Lynch — APPROVED** [review]

**Pull request merged**